src/charm.py

```diff
@@ -294,9 +294,7 @@ def get_secret(self, scope: Scopes, key: str) -> Optional[str]:
             return None
         secret_key = self._translate_field_to_secret_key(key)
         # Old translation in databag is to be taken
-        if key != secret_key and (
-            result := self.peer_relation_data(scope).fetch_my_relation_field(peers.id, key)
-        ):
```

Comment:
If key == secret_key we ignored that peer data value

Reply:
I'm a bit confused: If key == secret_key, then this means we should check inside the secret, not the databag (wasn't this the original intention of this code?). Then how come we can have key == secret_key and still have to check the databag? Doesn´t this mean something went wrong when setting the secret in the first place? Sorry I think I never understood this workflow properly :grimacing:

Reply:
I suppose the bug itself is not clear to me.

Reply:
Migrating from Juju 2 to Juju 3 would switch the charm from trying to use secrets, but all the values are still in the peer databag. The issue here is that we only check if the `key` and `secret_key` are different, but the data would still be in the peer databag even if the key is the same (no translation needed).

Reply:
Thanks for the explanation. Looks like a specific edge case for which we didn´t have any tests.